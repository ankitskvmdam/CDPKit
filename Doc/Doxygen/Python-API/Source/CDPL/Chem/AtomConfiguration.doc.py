#
# This file is part of the Chemical Data Processing Toolkit
#
# Copyright (C) Thomas Seidel <thomas.seidel@univie.ac.at>
#
# This program is free software; you can redistribute it and/or
# modify it under the terms of the GNU Lesser General Public
# License as published by the Free Software Foundation; either
# version 2 of the License, or (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
# Lesser General Public License for more details.
#
# You should have received a copy of the GNU Lesser General Public License
# along with this program; see the file COPYING. If not, write to
# the Free Software Foundation, Inc., 59 Temple Place - Suite 330,
# Boston, MA 02111-1307, USA.
#

##
# \brief Provides constants that are used to specify the <em>R/S</em> configuration of atoms.
# 
class AtomConfiguration(Boost.Python.instance):

    ##
    # \brief Specifies that the configuration of the atom is undefined.
    # 
    UNDEF = 0

    ##
    # \brief Specifies that the atom is not a stereogenic center and thus cannot be assigned a configuration.
    # 
    NONE = 1

    ##
    # \brief Specifies that the atom has <em>R</em> configuration.
    # 
    R = 2

    ##
    # \brief Specifies that the atom has <em>S</em> configuration.
    # 
    S = 4

    ##
    # \brief Specifies that the atom is a stereogenic center but has no defined configuration.
    # 
    EITHER = 8

    ##
    # \brief Specifies that the atom has <em>r</em> configuration.
    # 
    r = 16

    ##
    # \brief Specifies that the atom has <em>s</em> configuration.
    # 
    s = 17

    ##
    # \brief Specifies that the atom has <em>M</em> configuration.
    # 
    M = 19

    ##
    # \brief Specifies that the atom has <em>P</em> configuration.
    # 
    P = 18

    ##
    # \brief Specifies that the atom has <em>m</em> configuration.
    # 
    m = 21

    ##
    # \brief Specifies that the atom has <em>p</em> configuration.
    # 
    p = 20

    ##
    # \brief Specifies that the geometry of the ligand arrangement is <em>square planar</em>.
    # 
    SP = 22

    ##
    # \brief Specifies that the geometry of the ligand arrangement is <em>trigonal bipyramidal</em>.
    # 
    TB = 26

    ##
    # \brief Specifies that the geometry of the ligand arrangement is <em>octahedral</em>.
    # 
    OH = 47
