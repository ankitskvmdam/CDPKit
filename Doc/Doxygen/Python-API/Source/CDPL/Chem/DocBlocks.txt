#
# --- Unresolved property getter/setter functions ---
#
