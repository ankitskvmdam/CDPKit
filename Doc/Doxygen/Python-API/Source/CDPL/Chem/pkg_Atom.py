#
# This file is part of the Chemical Data Processing Toolkit
#
# Copyright (C) Thomas A. Seidel <thomas.seidel@univie.ac.at>
#
# This program is free software; you can redistribute it and/or
# modify it under the terms of the GNU Lesser General Public
# License as published by the Free Software Foundation; either
# version 2 of the License, or (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
# Lesser General Public License for more details.
#
# You should have received a copy of the GNU Lesser General Public License
# along with this program; see the file COPYING. If not, write to
# the Free Software Foundation, Inc., 59 Temple Place - Suite 330,
# Boston, MA 02111-1307, USA.
#

##
# \brief 
#
class Atom(AtomContainer, BondContainer, Entity3D):

    ##
    # \brief 
    #
    class AtomSequence(Boost.Python.instance):

        ##
        # \brief 
        # \return 
        #
        def __len__() -> int: pass

        ##
        # \brief 
        # \param idx 
        # \return 
        #
        def __getitem__(idx: int) -> Atom: pass

        ##
        # \brief Returns the result of the membership test operation <tt>atom in self</tt>.
        # \param atom The value to test for membership.
        # \return The result of the membership test operation.
        #
        def __contains__(atom: Atom) -> bool: pass

    ##
    # \brief 
    #
    class BondSequence(Boost.Python.instance):

        ##
        # \brief 
        # \return 
        #
        def __len__() -> int: pass

        ##
        # \brief 
        # \param idx 
        # \return 
        #
        def __getitem__(idx: int) -> Bond: pass

        ##
        # \brief Returns the result of the membership test operation <tt>bond in self</tt>.
        # \param bond The value to test for membership.
        # \return The result of the membership test operation.
        #
        def __contains__(bond: Bond) -> bool: pass

    ##
    # \brief Initializes the \e %Atom instance.
    #
    def __init__() -> None: pass

    ##
    # \brief 
    # \return 
    #
    def getMolecule() -> Molecule: pass

    ##
    # \brief 
    # \param atom 
    # \return 
    #
    def getBondToAtom(atom: Atom) -> Bond: pass

    ##
    # \brief 
    # \param atom 
    # \return 
    #
    def findBondToAtom(atom: Atom) -> Bond: pass

    ##
    # \brief 
    # \return 
    #
    def getIndex() -> int: pass

    ##
    # \brief Replaces the current state of \a self with a copy of the state of the \e %Atom instance \a atom.
    # \param atom The \e %Atom instance to copy.
    # \return The assignment target \a self.
    #
    def assign(atom: Atom) -> Atom: pass

    ##
    # \brief 
    # \return 
    #
    def getAtoms() -> object: pass

    ##
    # \brief 
    # \return 
    #
    def getBonds() -> object: pass

    ##
    # \brief 
    # \param idx 
    # \return 
    #
    def getAtom(idx: int) -> Atom: pass

    ##
    # \brief 
    # \param atom 
    # \return 
    #
    def containsAtom(atom: Atom) -> bool: pass

    ##
    # \brief 
    # \param arg1 
    # \param arg2 
    #
    def containsAtom(arg2: Atom) -> None: pass

    ##
    # \brief 
    # \param atom 
    # \return 
    #
    def getAtomIndex(atom: Atom) -> int: pass

    ##
    # \brief 
    # \param arg1 
    # \param arg2 
    #
    def getAtomIndex(arg2: Atom) -> None: pass

    ##
    # \brief 
    # \return 
    #
    def getNumAtoms() -> int: pass

    ##
    # \brief 
    # \param func 
    #
    def orderAtoms(func: CDPL.ForceField.InteractionFilterFunction2) -> None: pass

    ##
    # \brief 
    # \param idx 
    # \return 
    #
    def getEntity(idx: int) -> Entity3D: pass

    ##
    # \brief 
    # \return 
    #
    def getNumEntities() -> int: pass

    ##
    # \brief 
    # \param idx 
    # \return 
    #
    def getBond(idx: int) -> Bond: pass

    ##
    # \brief 
    # \param bond 
    # \return 
    #
    def containsBond(bond: Bond) -> bool: pass

    ##
    # \brief 
    # \param arg1 
    # \param arg2 
    #
    def containsBond(arg2: Bond) -> None: pass

    ##
    # \brief 
    # \param func 
    #
    def orderBonds(func: BoolBond2Functor) -> None: pass

    ##
    # \brief 
    # \param bond 
    # \return 
    #
    def getBondIndex(bond: Bond) -> int: pass

    ##
    # \brief 
    # \param arg1 
    # \param arg2 
    #
    def getBondIndex(arg2: Bond) -> None: pass

    ##
    # \brief 
    # \return 
    #
    def getNumBonds() -> int: pass

    ##
    # \brief 
    # \param key 
    # \param value 
    #
    def setProperty(key: CDPL.Base.LookupKey, value: CDPL.Base.Variant) -> None: pass

    ##
    # \brief 
    # \param key 
    # \return 
    #
    def removeProperty(key: CDPL.Base.LookupKey) -> bool: pass

    ##
    # \brief 
    # \param key 
    # \param throw_ 
    # \return 
    #
    def getProperty(key: CDPL.Base.LookupKey, throw_: bool = False) -> CDPL.Base.Variant: pass

    ##
    # \brief 
    # \param key 
    # \return 
    #
    def isPropertySet(key: CDPL.Base.LookupKey) -> bool: pass

    ##
    # \brief 
    #
    def clearProperties() -> None: pass

    ##
    # \brief 
    # \param cntnr 
    #
    def addProperties(cntnr: CDPL.Base.PropertyContainer) -> None: pass

    ##
    # \brief 
    # \param cntnr 
    #
    def copyProperties(cntnr: CDPL.Base.PropertyContainer) -> None: pass

    ##
    # \brief 
    # \param cntnr 
    #
    def swap(cntnr: CDPL.Base.PropertyContainer) -> None: pass

    ##
    # \brief Returns the result of the membership test operation <tt>atom in self</tt>.
    # \param atom The value to test for membership.
    # \return The result of the membership test operation.
    #
    def __contains__(atom: Atom) -> bool: pass

    ##
    # \brief Returns the result of the membership test operation <tt>bond in self</tt>.
    # \param bond The value to test for membership.
    # \return The result of the membership test operation.
    #
    def __contains__(bond: Bond) -> bool: pass

    ##
    # \brief Returns the result of the membership test operation <tt>key in self</tt>.
    # \param key The value to test for membership.
    # \return The result of the membership test operation.
    #
    def __contains__(key: CDPL.Base.LookupKey) -> bool: pass

    ##
    # \brief 
    # \param key 
    # \return 
    #
    def __getitem__(key: CDPL.Base.LookupKey) -> CDPL.Base.Variant: pass

    ##
    # \brief 
    # \param key 
    # \param value 
    #
    def __setitem__(key: CDPL.Base.LookupKey, value: CDPL.Base.Variant) -> None: pass

    ##
    # \brief 
    # \param key 
    # \return 
    #
    def __delitem__(key: CDPL.Base.LookupKey) -> bool: pass

    ##
    # \brief 
    # \return 
    #
    def __len__() -> int: pass

    ##
    # \brief 
    #
    molecule = property(getMolecule)

    ##
    # \brief 
    #
    index = property(getIndex)

    ##
    # \brief 
    #
    atoms = property(getAtoms)

    ##
    # \brief 
    #
    bonds = property(getBonds)
