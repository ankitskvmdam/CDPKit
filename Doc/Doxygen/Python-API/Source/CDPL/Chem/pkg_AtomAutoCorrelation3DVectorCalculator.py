#
# This file is part of the Chemical Data Processing Toolkit
#
# Copyright (C) Thomas A. Seidel <thomas.seidel@univie.ac.at>
#
# This program is free software; you can redistribute it and/or
# modify it under the terms of the GNU Lesser General Public
# License as published by the Free Software Foundation; either
# version 2 of the License, or (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
# Lesser General Public License for more details.
#
# You should have received a copy of the GNU Lesser General Public License
# along with this program; see the file COPYING. If not, write to
# the Free Software Foundation, Inc., 59 Temple Place - Suite 330,
# Boston, MA 02111-1307, USA.
#

##
# \brief 
#
class AtomAutoCorrelation3DVectorCalculator(Boost.Python.instance):

    ##
    # \brief Initializes the \e %AtomAutoCorrelation3DVectorCalculator instance.
    # \param self The \e %AtomAutoCorrelation3DVectorCalculator instance to initialize.
    #
    def __init__(self: object) -> None: pass

    ##
    # \brief Initializes the \e %AtomAutoCorrelation3DVectorCalculator instance.
    # \param self The \e %AtomAutoCorrelation3DVectorCalculator instance to initialize.
    # \param calculator 
    #
    def __init__(self: object, calculator: AtomAutoCorrelation3DVectorCalculator) -> None: pass

    ##
    # \brief Initializes the \e %AtomAutoCorrelation3DVectorCalculator instance.
    # \param self The \e %AtomAutoCorrelation3DVectorCalculator instance to initialize.
    # \param cntnr 
    # \param vec 
    #
    def __init__(self: object, cntnr: AtomContainer, vec: CDPL.Math.DVector) -> None: pass

    ##
    # \brief Returns the numeric identifier (ID) of the wrapped C++ class instance.
    # \param self The \e %AtomAutoCorrelation3DVectorCalculator instance this method is called upon.
    #
    # Different Python \e %AtomAutoCorrelation3DVectorCalculator instances may reference the same underlying C++ class instance. The commonly used Python expression
    # <tt>a is not b</tt> thus cannot tell reliably whether the two \e %AtomAutoCorrelation3DVectorCalculator instances \e a and \e b reference different C++ objects. 
    # The numeric identifier returned by this method allows to correctly implement such an identity test via the simple expression
    # <tt>a.getObjectID() != b.getObjectID()</tt>.
    #
    # \return The numeric ID of the internally referenced C++ class instance.
    #
    def getObjectID(self: AtomAutoCorrelation3DVectorCalculator) -> int: pass

    ##
    # \brief Replaces the current state of \a self with a copy of the state of the \e %AtomAutoCorrelation3DVectorCalculator instance \a calculator.
    # \param self The \e %AtomAutoCorrelation3DVectorCalculator instance this method is called upon.
    # \param calculator The \e %AtomAutoCorrelation3DVectorCalculator instance to copy.
    # \return The assignment target \a self.
    #
    def assign(self: AtomAutoCorrelation3DVectorCalculator, calculator: AtomAutoCorrelation3DVectorCalculator) -> AtomAutoCorrelation3DVectorCalculator: pass

    ##
    # \brief 
    # \param self The \e %AtomAutoCorrelation3DVectorCalculator instance this method is called upon.
    # \param func 
    #
    def setEntityPairWeightFunction(self: AtomAutoCorrelation3DVectorCalculator, func: DoubleAtom2Functor) -> None: pass

    ##
    # \brief 
    # \param self The \e %AtomAutoCorrelation3DVectorCalculator instance this method is called upon.
    # \param func 
    #
    def setEntity3DCoordinatesFunction(self: AtomAutoCorrelation3DVectorCalculator, func: Atom3DCoordinatesFunction) -> None: pass

    ##
    # \brief 
    # \param self The \e %AtomAutoCorrelation3DVectorCalculator instance this method is called upon.
    # \param num_steps 
    #
    def setNumSteps(self: AtomAutoCorrelation3DVectorCalculator, num_steps: int) -> None: pass

    ##
    # \brief 
    # \param self The \e %AtomAutoCorrelation3DVectorCalculator instance this method is called upon.
    # \return 
    #
    def getNumSteps(self: AtomAutoCorrelation3DVectorCalculator) -> int: pass

    ##
    # \brief 
    # \param self The \e %AtomAutoCorrelation3DVectorCalculator instance this method is called upon.
    # \param radius_inc 
    #
    def setRadiusIncrement(self: AtomAutoCorrelation3DVectorCalculator, radius_inc: float) -> None: pass

    ##
    # \brief 
    # \param self The \e %AtomAutoCorrelation3DVectorCalculator instance this method is called upon.
    # \return 
    #
    def getRadiusIncrement(self: AtomAutoCorrelation3DVectorCalculator) -> float: pass

    ##
    # \brief 
    # \param self The \e %AtomAutoCorrelation3DVectorCalculator instance this method is called upon.
    # \param start_radius 
    #
    def setStartRadius(self: AtomAutoCorrelation3DVectorCalculator, start_radius: float) -> None: pass

    ##
    # \brief 
    # \param self The \e %AtomAutoCorrelation3DVectorCalculator instance this method is called upon.
    # \return 
    #
    def getStartRadius(self: AtomAutoCorrelation3DVectorCalculator) -> float: pass

    ##
    # \brief 
    # \param self The \e %AtomAutoCorrelation3DVectorCalculator instance this method is called upon.
    # \param cntnr 
    # \param vec 
    #
    def calculate(self: AtomAutoCorrelation3DVectorCalculator, cntnr: AtomContainer, vec: CDPL.Math.DVector) -> None: pass

    ##
    # \brief 
    #
    objectID = property(getObjectID)

    ##
    # \brief 
    #
    startRadius = property(getStartRadius, setStartRadius)

    ##
    # \brief 
    #
    radiusIncrement = property(getRadiusIncrement, setRadiusIncrement)

    ##
    # \brief 
    #
    numSteps = property(getNumSteps, setNumSteps)
