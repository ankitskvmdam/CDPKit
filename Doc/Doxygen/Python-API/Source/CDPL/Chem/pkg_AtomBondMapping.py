#
# This file is part of the Chemical Data Processing Toolkit
#
# Copyright (C) Thomas A. Seidel <thomas.seidel@univie.ac.at>
#
# This program is free software; you can redistribute it and/or
# modify it under the terms of the GNU Lesser General Public
# License as published by the Free Software Foundation; either
# version 2 of the License, or (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
# Lesser General Public License for more details.
#
# You should have received a copy of the GNU Lesser General Public License
# along with this program; see the file COPYING. If not, write to
# the Free Software Foundation, Inc., 59 Temple Place - Suite 330,
# Boston, MA 02111-1307, USA.
#

##
# \brief 
#
class AtomBondMapping(Boost.Python.instance):

    ##
    # \brief Initializes the \e %AtomBondMapping instance.
    #
    def __init__() -> None: pass

    ##
    # \brief Initializes the \e %AtomBondMapping instance.
    # \param mapping 
    #
    def __init__(mapping: AtomBondMapping) -> None: pass

    ##
    # \brief Returns the numeric identifier (ID) of the wrapped C++ class instance.
    #
    # Different Python \e %AtomBondMapping instances may reference the same underlying C++ class instance. The commonly used Python expression
    # <tt>a is not b</tt> thus cannot tell reliably whether the two \e %AtomBondMapping instances \e a and \e b reference different C++ objects. 
    # The numeric identifier returned by this method allows to correctly implement such an identity test via the simple expression
    # <tt>a.getObjectID() != b.getObjectID()</tt>.
    #
    # \return The numeric ID of the internally referenced C++ class instance.
    #
    def getObjectID() -> int: pass

    ##
    # \brief Replaces the current state of \a self with a copy of the state of the \e %AtomBondMapping instance \a mapping.
    # \param mapping The \e %AtomBondMapping instance to copy.
    # \return The assignment target \a self.
    #
    def assign(mapping: AtomBondMapping) -> AtomBondMapping: pass

    ##
    # \brief 
    # \return 
    #
    def getAtomMapping() -> AtomMapping: pass

    ##
    # \brief 
    # \return 
    #
    def getBondMapping() -> BondMapping: pass

    ##
    # \brief 
    #
    def clear() -> None: pass

    ##
    # \brief Returns the result of the comparison operation <tt>self == mapping</tt>.
    # \param mapping The \e %AtomBondMapping instance to be compared with.
    # \return The result of the comparison operation.
    #
    def __eq__(mapping: AtomBondMapping) -> bool: pass

    ##
    # \brief Returns the result of the comparison operation <tt>self != mapping</tt>.
    # \param mapping The \e %AtomBondMapping instance to be compared with.
    # \return The result of the comparison operation.
    #
    def __ne__(mapping: AtomBondMapping) -> bool: pass

    ##
    # \brief 
    #
    objectID = property(getObjectID)

    ##
    # \brief 
    #
    atomMapping = property(getAtomMapping)

    ##
    # \brief 
    #
    bondMapping = property(getBondMapping)
