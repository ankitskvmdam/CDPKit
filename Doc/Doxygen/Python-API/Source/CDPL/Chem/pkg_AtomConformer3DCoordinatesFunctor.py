#
# This file is part of the Chemical Data Processing Toolkit
#
# Copyright (C) Thomas A. Seidel <thomas.seidel@univie.ac.at>
#
# This program is free software; you can redistribute it and/or
# modify it under the terms of the GNU Lesser General Public
# License as published by the Free Software Foundation; either
# version 2 of the License, or (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
# Lesser General Public License for more details.
#
# You should have received a copy of the GNU Lesser General Public License
# along with this program; see the file COPYING. If not, write to
# the Free Software Foundation, Inc., 59 Temple Place - Suite 330,
# Boston, MA 02111-1307, USA.
#

##
# \brief 
#
class AtomConformer3DCoordinatesFunctor(Boost.Python.instance):

    ##
    # \brief Initializes the \e %AtomConformer3DCoordinatesFunctor instance.
    # \param self The \e %AtomConformer3DCoordinatesFunctor instance to initialize.
    # \param func 
    #
    def __init__(self: object, func: AtomConformer3DCoordinatesFunctor) -> None: pass

    ##
    # \brief Initializes the \e %AtomConformer3DCoordinatesFunctor instance.
    # \param self The \e %AtomConformer3DCoordinatesFunctor instance to initialize.
    # \param conf_idx 
    #
    def __init__(self: object, conf_idx: int) -> None: pass

    ##
    # \brief Returns the numeric identifier (ID) of the wrapped C++ class instance.
    # \param self The \e %AtomConformer3DCoordinatesFunctor instance this method is called upon.
    #
    # Different Python \e %AtomConformer3DCoordinatesFunctor instances may reference the same underlying C++ class instance. The commonly used Python expression
    # <tt>a is not b</tt> thus cannot tell reliably whether the two \e %AtomConformer3DCoordinatesFunctor instances \e a and \e b reference different C++ objects. 
    # The numeric identifier returned by this method allows to correctly implement such an identity test via the simple expression
    # <tt>a.getObjectID() != b.getObjectID()</tt>.
    #
    # \return The numeric ID of the internally referenced C++ class instance.
    #
    def getObjectID(self: AtomConformer3DCoordinatesFunctor) -> int: pass

    ##
    # \brief Replaces the current state of \a self with a copy of the state of the \e %AtomConformer3DCoordinatesFunctor instance \a func.
    # \param self The \e %AtomConformer3DCoordinatesFunctor instance this method is called upon.
    # \param func The \e %AtomConformer3DCoordinatesFunctor instance to copy.
    # \return The assignment target \a self.
    #
    def assign(self: AtomConformer3DCoordinatesFunctor, func: AtomConformer3DCoordinatesFunctor) -> AtomConformer3DCoordinatesFunctor: pass

    ##
    # \brief 
    # \param self The \e %AtomConformer3DCoordinatesFunctor instance this method is called upon.
    # \param atom 
    # \return 
    #
    def __call__(self: AtomConformer3DCoordinatesFunctor, atom: Atom) -> CDPL.Math.Vector3D: pass

    ##
    # \brief 
    #
    objectID = property(getObjectID)
