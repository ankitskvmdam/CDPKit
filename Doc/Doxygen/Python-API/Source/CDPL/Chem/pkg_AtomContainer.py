#
# This file is part of the Chemical Data Processing Toolkit
#
# Copyright (C) Thomas Seidel <thomas.seidel@univie.ac.at>
#
# This program is free software; you can redistribute it and/or
# modify it under the terms of the GNU Lesser General Public
# License as published by the Free Software Foundation; either
# version 2 of the License, or (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
# Lesser General Public License for more details.
#
# You should have received a copy of the GNU Lesser General Public License
# along with this program; see the file COPYING. If not, write to
# the Free Software Foundation, Inc., 59 Temple Place - Suite 330,
# Boston, MA 02111-1307, USA.
#

##
# \brief 
#
class AtomContainer(Entity3DContainer):

    ##
    # \brief Initializes the \e %AtomContainer instance.
    #
    def __init__() -> None: pass

    ##
    # \brief 
    # \param idx 
    # \return 
    #
    def getAtom(idx: int) -> Atom: pass

    ##
    # \brief 
    # \param atom 
    # \return 
    #
    def containsAtom(atom: Atom) -> bool: pass

    ##
    # \brief 
    # \param atom 
    # \return 
    #
    def getAtomIndex(atom: Atom) -> int: pass

    ##
    # \brief 
    # \return 
    #
    def getNumAtoms() -> int: pass

    ##
    # \brief 
    # \param func 
    #
    def orderAtoms(func: CDPL.ForceField.InteractionFilterFunction2) -> None: pass

    ##
    # \brief 
    # \param idx 
    # \return 
    #
    def getEntity(idx: int) -> Entity3D: pass

    ##
    # \brief 
    # \return 
    #
    def getNumEntities() -> int: pass

    ##
    # \brief Returns the result of the membership test operation <tt>atom in self</tt>.
    # \param atom The value to test for membership.
    # \return The result of the membership test operation.
    #
    def __contains__(atom: Atom) -> bool: pass

    ##
    # \brief 
    # \param idx 
    # \return 
    #
    def __getitem__(idx: int) -> Atom: pass

    ##
    # \brief 
    # \return 
    #
    def __len__() -> int: pass

    ##
    # \brief 
    #
    numAtoms = property(getNumAtoms)
