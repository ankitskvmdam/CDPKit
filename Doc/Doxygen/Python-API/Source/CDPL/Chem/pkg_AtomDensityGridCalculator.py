#
# This file is part of the Chemical Data Processing Toolkit
#
# Copyright (C) Thomas A. Seidel <thomas.seidel@univie.ac.at>
#
# This program is free software; you can redistribute it and/or
# modify it under the terms of the GNU Lesser General Public
# License as published by the Free Software Foundation; either
# version 2 of the License, or (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
# Lesser General Public License for more details.
#
# You should have received a copy of the GNU Lesser General Public License
# along with this program; see the file COPYING. If not, write to
# the Free Software Foundation, Inc., 59 Temple Place - Suite 330,
# Boston, MA 02111-1307, USA.
#

##
# \brief 
#
class AtomDensityGridCalculator(Boost.Python.instance):

    ##
    # \brief Initializes the \e %AtomDensityGridCalculator instance.
    # \param self The \e %AtomDensityGridCalculator instance to initialize.
    #
    def __init__(self: object) -> None: pass

    ##
    # \brief Initializes the \e %AtomDensityGridCalculator instance.
    # \param self The \e %AtomDensityGridCalculator instance to initialize.
    # \param func 
    #
    def __init__(self: object, func: DoubleVector3D2AtomFunctor) -> None: pass

    ##
    # \brief Initializes the \e %AtomDensityGridCalculator instance.
    # \param self The \e %AtomDensityGridCalculator instance to initialize.
    # \param density_func 
    # \param comb_func 
    #
    def __init__(self: object, density_func: DoubleVector3D2AtomFunctor, comb_func: CDPL.Math.DoubleDVectorFunctor) -> None: pass

    ##
    # \brief Initializes the \e %AtomDensityGridCalculator instance.
    # \param self The \e %AtomDensityGridCalculator instance to initialize.
    # \param calculator 
    #
    def __init__(self: object, calculator: AtomDensityGridCalculator) -> None: pass

    ##
    # \brief Returns the numeric identifier (ID) of the wrapped C++ class instance.
    # \param self The \e %AtomDensityGridCalculator instance this method is called upon.
    #
    # Different Python \e %AtomDensityGridCalculator instances may reference the same underlying C++ class instance. The commonly used Python expression
    # <tt>a is not b</tt> thus cannot tell reliably whether the two \e %AtomDensityGridCalculator instances \e a and \e b reference different C++ objects. 
    # The numeric identifier returned by this method allows to correctly implement such an identity test via the simple expression
    # <tt>a.getObjectID() != b.getObjectID()</tt>.
    #
    # \return The numeric ID of the internally referenced C++ class instance.
    #
    def getObjectID(self: AtomDensityGridCalculator) -> int: pass

    ##
    # \brief Replaces the current state of \a self with a copy of the state of the \e %AtomDensityGridCalculator instance \a calculator.
    # \param self The \e %AtomDensityGridCalculator instance this method is called upon.
    # \param calculator The \e %AtomDensityGridCalculator instance to copy.
    # \return The assignment target \a self.
    #
    def assign(self: AtomDensityGridCalculator, calculator: AtomDensityGridCalculator) -> AtomDensityGridCalculator: pass

    ##
    # \brief 
    # \param self The \e %AtomDensityGridCalculator instance this method is called upon.
    # \param dist 
    #
    def setDistanceCutoff(self: AtomDensityGridCalculator, dist: float) -> None: pass

    ##
    # \brief 
    # \param self The \e %AtomDensityGridCalculator instance this method is called upon.
    # \return 
    #
    def getDistanceCutoff(self: AtomDensityGridCalculator) -> float: pass

    ##
    # \brief 
    # \param self The \e %AtomDensityGridCalculator instance this method is called upon.
    # \return 
    #
    def getDensityFunction(self: AtomDensityGridCalculator) -> DoubleVector3D2AtomFunctor: pass

    ##
    # \brief 
    # \param self The \e %AtomDensityGridCalculator instance this method is called upon.
    # \param func 
    #
    def setDensityFunction(self: AtomDensityGridCalculator, func: DoubleVector3D2AtomFunctor) -> None: pass

    ##
    # \brief 
    # \param self The \e %AtomDensityGridCalculator instance this method is called upon.
    # \param func 
    #
    def setDensityCombinationFunction(self: AtomDensityGridCalculator, func: CDPL.Math.DoubleDVectorFunctor) -> None: pass

    ##
    # \brief 
    # \param self The \e %AtomDensityGridCalculator instance this method is called upon.
    # \return 
    #
    def getDensityCombinationFunction(self: AtomDensityGridCalculator) -> CDPL.Math.DoubleDVectorFunctor: pass

    ##
    # \brief 
    # \param self The \e %AtomDensityGridCalculator instance this method is called upon.
    # \param func 
    #
    def setAtom3DCoordinatesFunction(self: AtomDensityGridCalculator, func: Atom3DCoordinatesFunction) -> None: pass

    ##
    # \brief 
    # \param self The \e %AtomDensityGridCalculator instance this method is called upon.
    # \return 
    #
    def getAtom3DCoordinatesFunction(self: AtomDensityGridCalculator) -> Atom3DCoordinatesFunction: pass

    ##
    # \brief 
    # \param self The \e %AtomDensityGridCalculator instance this method is called upon.
    # \param atoms 
    # \param grid 
    #
    def calculate(self: AtomDensityGridCalculator, atoms: AtomContainer, grid: CDPL.Grid.DSpatialGrid) -> None: pass

    ##
    # \brief 
    #
    objectID = property(getObjectID)

    ##
    # \brief 
    #
    distanceCutoff = property(getDistanceCutoff, setDistanceCutoff)

    ##
    # \brief 
    #
    densityFunction = property(getDensityFunction, setDensityFunction)

    ##
    # \brief 
    #
    densityCombinationFunction = property(getDensityCombinationFunction, setDensityCombinationFunction)

    ##
    # \brief FIXME!
    #
    atomCoordinatesFunction = property(getAtomCoordinatesFunction, setAtomCoordinatesFunction)
