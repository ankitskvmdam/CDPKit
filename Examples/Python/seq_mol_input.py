#!/bin/env python

##
# seq_mol_input.py 
#
# This file is part of the Chemical Data Processing Toolkit
#
# Copyright (C) 2003-2022 Thomas A. Seidel <thomas.seidel@univie.ac.at>
#
# This program is free software; you can redistribute it and/or
# modify it under the terms of the GNU Lesser General Public
# License as published by the Free Software Foundation; either
# version 2 of the License, or (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
# Lesser General Public License for more details.
#
# You should have received a copy of the GNU Lesser General Public License
# along with this program; see the file COPYING. If not, write to
# the Free Software Foundation, Inc., 59 Temple Place - Suite 330,
# Boston, MA 02111-1307, USA.
##


import sys

import CDPL.Chem as Chem


# function called for each read molecule
def procMolecule(mol: Chem.Molecule) -> None: 
    print('Processing molecule consisting of {!s} atoms and {!s} bonds'.format(mol.numAtoms, mol.numBonds))

def main() -> None:
    if len(sys.argv) < 2:
        sys.exit('Usage: %s <input SD-file>' % sys.argv[0])

    # create a reader instance for SD-files providing the filename as constructor argument
    reader = Chem.FileSDFMoleculeReader(sys.argv[1])

    # create an instance of the default implementation of the Molecule interface
    mol = Chem.BasicMolecule()
    
    # read and process molecules one after the other until the end of input has been reached
    try:
        while reader.read(mol): 
            procMolecule(mol)
    except: # handle exception raised in case of severe read errors
        print('Error while reading molecule!', file=sys.stderr)
        return
        
if __name__ == '__main__':
    main()
