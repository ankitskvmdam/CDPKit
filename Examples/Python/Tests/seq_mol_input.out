Processing molecule consisting of 45 atoms and 47 bonds
